### PR TITLE
Clean up event loop from client

### PR DESCRIPTION
**Pull request by WyriHaximus**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

**Comment by dbu** [comment]
thanks. i guess we don't care about the roave BC break complaint - the 4.x branch is allowed to have BC breaks. but can you explicitly mention this in the changelog?

there seems to be a place where loop is still referenced in the Client.php, according to phpstan. can you check that please?

and can you please again do the cs fix?  

i merged #56, so this can be rebased on 4.x to have the last state.

**Comment by WyriHaximus** [comment]
Clean all failing checks up before opening it for review

**Comment by WyriHaximus** [comment]
> thanks. i guess we don't care about the roave BC break complaint - the 4.x branch is allowed to have BC breaks. but can you explicitly mention this in the changelog?

Yes let me update this PR with an explanation in the changelog before you merged it

> there seems to be a place where loop is still referenced in the Client.php, according to phpstan. can you check that please?
> 
> and can you please again do the cs fix?
> 
> i merged #56, so this can be rebased on 4.x to have the last state.

Done, done, and done.

**Comment by WyriHaximus** [comment]
~Also, where is the documentation for this package located? Will also update those while I'm at it :).~

Nvm it's at the top of the page at https://docs.php-http.org/en/latest/clients/react-adapter.html

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,8 +25,19 @@ async(static function () {
 })();
 ```
 
+Another major change in this release is that injecting the event loop has been removed, and now fully uses
```
Comment (dbu) [inline CHANGELOG.md]:
```suggestion
Another major change in this release is that you no longer inject the event loop into the client. It now only uses
```

Comment (WyriHaximus) [reply CHANGELOG.md]: Updated 👍

**Comment by dbu** [comment]
shall i tag now, or do you plan other changes?

**Comment by WyriHaximus** [comment]
Go for it, I'm currently updating the documentation, which seems to be "somewhat" out of date 😅

**Comment by dbu** [comment]
here we go: https://github.com/php-http/react-adapter/releases/tag/4.0.0

do you want commit rights on the react-adapter so you can create releases and merge changes? i am happy to keep doing reviews for general sanity and understandability, but have very little experience with reactphp and event loops.

**Comment by WyriHaximus** [comment]
> here we go: https://github.com/php-http/react-adapter/releases/tag/4.0.0

🎉 

> do you want commit rights on the react-adapter so you can create releases and merge changes? i am happy to keep doing reviews for general sanity and understandability, but have very little experience with reactphp and event loops.

Sure :+1:

**Comment by dbu** [comment]
great, thanks for this. i sent you the invite.

**Pull request merged**